tests/test_metrics_api.py

```diff
@@ -0,0 +1,362 @@
+
+    @pytest.fixture(autouse=True)
+    def setup_post(self, _build_snql_post_methods: Callable[[str], Any]) -> None:
+        self.post = _build_snql_post_methods
+
+    def setup_method(self, test_method: Any) -> None:
+        super().setup_method(test_method)
+
+        # values for test data
+        self.metric_id = 1001
+        self.org_id = 101
+        self.project_ids = [1, 2]  # 2 projects
+        self.seconds = 180 * 60
+
+        self.default_tags = {
+            TAG_1_KEY: TAG_1_VALUE_1,
+            TAG_2_KEY: TAG_2_VALUE_1,
+            TAG_3_KEY: TAG_3_VALUE_1,
+            TAG_4_KEY: TAG_4_VALUE_1,
+        }
+        self.skew = timedelta(seconds=self.seconds)
+
+        self.base_time = datetime.utcnow().replace(
+            minute=0, second=0, microsecond=0, tzinfo=pytz.utc
+        ) - timedelta(minutes=self.seconds)
+        self.storage = get_writable_storage(StorageKey.METRICS_COUNTERS_BUCKETS)
+        self.generate_counters()
+
+    def teardown_method(self, test_method: Any) -> None:
+        teardown_common()
+
+    def generate_counters(self) -> None:
+        events = []
+        for n in range(self.seconds):
+            for p in self.project_ids:
+                processed = (
+                    self.storage.get_table_writer()
+                    .get_stream_loader()
+                    .get_processor()
+                    .process_message(
+                        (
+                            {
+                                "org_id": self.org_id,
+                                "project_id": p,
+                                "unit": "ms",
+                                "type": METRICS_COUNTERS_TYPE,
+                                "value": 1.0,
+                                "timestamp": self.base_time.timestamp() + n,
+                                "tags": self.default_tags,
+                                "metric_id": self.metric_id,
+                                "retention_days": RETENTION_DAYS,
+                            }
+                        ),
+                        KafkaMessageMetadata(0, 0, self.base_time),
+                    )
+                )
+                if processed:
+                    events.append(processed)
+        write_processed_messages(self.storage, events)
+
+    def build_simple_query(
+        self,
+        metric_id: Optional[int] = None,
+        org_id: Optional[int] = None,
+        start_time: Optional[str] = None,
+        end_time: Optional[str] = None,
+    ) -> str:
+        if not metric_id:
+            metric_id = self.metric_id
+        if not org_id:
+            org_id = self.org_id
+        if not start_time:
+            start_time = (self.base_time - self.skew).isoformat()
+        if not end_time:
+            end_time = (self.base_time + self.skew).isoformat()
+        query_str = """MATCH (metrics_counters)
+                    SELECT sumMerge(value) AS total_seconds BY project_id, org_id
```

Comment:
does the match query need to use `sumMerge` ? Do the translation mappers https://github.com/getsentry/snuba/blob/master/snuba/datasets/entities/metrics.py#L151 handle converting that?

Reply:
I didn't know those existed, and I think I should definitely use them, as that tests more of the end-to-end for these datasets/entities